test/inductor/test_torchinductor_opinfo.py

```diff
@@ -322,6 +322,72 @@ def wrapper_noop_set_seed(op, *args, **kwargs):
 )
 
 
+def argsort_equal_checker(
+    self: TestCase,
+    ref_inputs,
+    example_inputs,
+    correct,
+    actual,
+    *,
+    atol=None,
+    rtol=None,
+    exact_dtype=True,
+):
+    is_scalar = len(ref_inputs[0].shape) == 0
+
+    if not is_scalar:
+        self.assertEqual(
+            ref_inputs[0][actual],
+            ref_inputs[0][correct],
```

Comment:
This is too loose for stable sort. Also can you explain why this fails? What is different between inductor and eager that causes the returned indices to be different?

Reply:
That comparison definitely isn't right for stable sorts, I agree. Are any sorts with stable=True being tested by that code? I wasn't able to tell. 
A simple equality check of the indices can fail if there are repeated values, since x86-simd-sort might give the indices of repeats in a different order compared to the reference. I'm not familiar with the inductor code, but from my inspecting it seems to use this Triton kernel https://github.com/pytorch/pytorch/blob/873abfc18e8946bde5c090dda1b44d21c46c0e46/torch/_inductor/runtime/triton_helpers.py#L506

Reply:
> A simple equality check of the indices can fail if there are repeated values, since x86-simd-sort might give the indices of repeats in a different order compared to the reference.

Triton is only used for CUDA, on CPU we should be comparing `at::sort` to `at::sort` here so I would expect the indices of repeated values to be the same. Otherwise it suggests the sort is non-deterministic.

Reply:
Okay, I've determined why this seemed so strange. In the check_model code, reference_in_float defaults to True, so it was converting the fp16 values to fp32 to generate the reference results; XSS sort can sort fp32 values, but not fp16, so the result was that the fp16 values were sorted using the normal sorting method, but the fp32 values were sorted using XSS, which sort equivalent values differently, causing the fp16 test failures seen before.